src/track/track.cpp

```diff
@@ -812,13 +812,39 @@ void Track::importCuePoints(const QList<mixxx::CueInfo>& cueInfos) {
         trackId = m_record.getId();
         sampleRate = m_record.getMetadata().getSampleRate();
     } // implicitly unlocked when leaving scope
+    DEBUG_ASSERT(sampleRate.valid());
 
     QList<CuePointer> cuePoints;
     for (const mixxx::CueInfo& cueInfo : cueInfos) {
         CuePointer pCue(new Cue(trackId, sampleRate, cueInfo));
         cuePoints.append(pCue);
     }
+
     setCuePoints(cuePoints);
+    kLogger.debug()
+            << "Imported"
+            << cuePoints.size()
+            << "cues for this track.";
+}
+
+void Track::importCuePointsLater(const QList<mixxx::CueInfo>& cueInfos) {
```

Comment:
@uklotzde Would this be a worthwhile approach to solve the problem of delaying import while the sample rate is unknown?

Reply:
A reasonable and pragmatic workaround. Loading cue points as POD `CueInfo` and converting them on the fly into `Cue` objects after the `SoundSource` of the track has been opened is clever. We might need this anyway if we decide to migrate our existing `Cue` data.

I still need to think more about the consequences and if this is the right place and time.